src/form/input-wrapper.js

```diff
@@ -101,39 +102,10 @@ export class InputWrapper extends Component {
     return render({
       value: this.formatNumber(),
       update: updateFormModel.bind(null, name, type || 'text'),
-      validate: validateField.bind(null, name),
       error: errors[name],
-      isValidating: validating[name],
-      formDisabled,
-      updateFormModel,
-      validateField,
-      showRealValue: this.showRealValue,
-      showPrettyValueAndValidate: this.showPrettyValueAndValidate
-    })
-  }
-}
-
-export class SubmitWrapper extends Component {
-  static propTypes = {
-    render: func.isRequired,
-  }
-
-  static contextTypes = {
-    form: object.isRequired,
-    formActions: object.isRequired,
-  }
-
-  render() {
-    const {
-      form: { formDisabled, formValidating, formValid },
-      formActions: { submitForm },
-    } = this.context
-
-    return this.props.render({
-      formDisabled,
-      formValidating,
-      formValid,
-      submitForm,
+      inputDisabled: validating[name] || formDisabled,
```

Comment:
we should add `formValidating` as an additional OR condition here (idea being if one input is validating asynchronously and that validation is dependent on the value of another input, the entire form should be disabled)